### ISSUE TITLE
How can I load contrib warts into IntelliJ IDEA?

### ISSUE THREAD

**Issue by umireon** [wartremover/wartremover-contrib#218]
## My question

I tried to use contrib warts on this repository on IntelliJ IDEA. Contrib warts were successfully loaded on SBT builds and it procuded wart violations with SBT. However, when I instructed IntelliJ IDEA to build Scala classes, the contrib warts were completely ignored by IntelliJ IDEA's compiler. Warts that are built in [Wartremover core](https://github.com/wartremover/wartremover) could be loaded into IntelliJ IDEA's compiler and it produced wart violation as expected. How can I load contrib warts into IntelliJ IDEA?

## My code

https://github.com/umireon/umireon-wartremover-contrib-intellij-idea/tree/intellij-idea-cannot-load-wartremover-contrib-classes

We prefer the `-Xfatal-warnings` flag and this problem also prevents us to use many IntelliJ IDEA features.

This problem causes the following error:

<img width="1624" alt="IntelliJ IDEA cannot load contrib warts" src="https://user-images.githubusercontent.com/1067855/187628885-01cece87-1b99-4013-9092-eb87af9a2167.png">

> scalac: load failure warts = (org.wartremover.contrib.warts.Apply,scala.ScalaReflectionException: object org.wartremover.contrib.warts.Apply not found.)

I checked the configuration of Scala compiler on Intellij IDEA and it revealed that it loaded Wartremover core but didn't load Wartremover contrib.

<img width="1094" alt="The Scala compile on IntelliJ IDEA only loads wartremover not wartremover-contrib" src="https://user-images.githubusercontent.com/1067855/187629327-c006b7a7-0d4a-4d98-ba72-fb6a934e754a.png">

Thanks in advance.

Best,

**Comment by tsawada** [comment]
@xuwei-k do you have any idea how to get contrib warts work in IntelliJ IDEA?

**Comment by hagay3** [comment]
@xuwei-k 
having same issue with intellij idea, I think it should be addressed somehow.
currently, it's not possible to use contrib wart with IntelliJ idea.
```
scalac: load failure warts = (org.wartremover.contrib.warts.DiscardedFuture,scala.ScalaReflectionException: object org.wartremover.contrib.warts.DiscardedFuture not found.)
```

**Comment by tsawada** [comment]
@hagay3 
I believe there's something missing in the IntelliJ Scala plugin. I filed an issue here: https://youtrack.jetbrains.com/issue/SCL-21154

**Comment by mvestola** [comment]
This seems to be fixed in IntelliJ IDEA latest 2023.3 EAP at least (which will be soon future 2023.3 release). I was using EAP version "Build #IU-233.9802.14, built on October 12, 2023". I got the same kind of failure with the EAP build: `scalac: load failure warts = (org.wartremover.contrib.warts.DiscardedFuture,scala.ScalaReflectionException: object org.wartremover.contrib.warts.OldTime not found.)` when running scala in IDEA. But updating Scala plugin to the latest one 2023.3.489 fixed the issue for me.

So basically, update to latest IDEA and update Scala plugin should make it work.

**Comment by mvestola** [comment]
> This seems to be fixed in IntelliJ IDEA latest 2023.3 EAP at least (which will be soon future 2023.3 release). I was using EAP version "Build #IU-233.9802.14, built on October 12, 2023". I got the same kind of failure with the EAP build: `scalac: load failure warts = (org.wartremover.contrib.warts.DiscardedFuture,scala.ScalaReflectionException: object org.wartremover.contrib.warts.OldTime not found.)` when running scala in IDEA. But updating Scala plugin to the latest one 2023.3.489 fixed the issue for me.
> 
> So basically, update to latest IDEA and update Scala plugin should make it work.

Seems that I was partly wrong. Still getting the same issue even with the latest Intellij IDEA releases. I have following line in my `build.sbt`:
```
wartremoverErrors += ContribWart.OldTime,
```
If I comment that out and run sbt update on IDEA (by clicking the sbt refresh icon) and then uncomment it and run sbt update again -> I get no errors. But after any dependency is updated and sbt update happens automatically by IDEA again, it suddenly starts producing the same error again. Seems really weird but definitely there is still some issue in the IDEA or IDEA's scala plugin...

**Comment by ncreep** [comment]
The workaround suggested [here](https://youtrack.jetbrains.com/issue/SCL-21154/wartremover-contrib-scalac-plugin-does-not-work-in-IntelliJ-IDEA#focus=Comments-27-7283130.0-0) seems to be working for me.
